client/network/src/behaviour.rs

```diff
@@ -45,6 +45,8 @@ pub struct Behaviour<B: BlockT, H: ExHashT> {
 	discovery: DiscoveryBehaviour,
 	/// Block request handling.
 	block_requests: protocol::BlockRequests<B>,
+	/// Finality proof request handling.
+	finality_proof_requests: protocol::FinalityProofRequests<B>,
```

Comment:
Would it make sense to use [`Toggle<protocol::FinalityProofRequests<B>>`](https://docs.rs/libp2p/0.18.1/libp2p/swarm/toggle/struct.Toggle.html) here instead of storing an `Option<Arc<dyn FinalityProofProvider<B>>>` in `FinalityProofRequests` and erroring on incoming requests if it is a `None`?

Reply:
I went for keeping the current behaviour because I didn't know when the `Option` can be `None`. But I think it is `None` for light clients (which can't serve finality proofs) and `Some` for full nodes.

I'll do the change.